packages/handlefi/src/input-filters.ts

```diff
@@ -0,0 +1,190 @@
+          fromAmount: amountIn,
+          toAmount: amountOut,
+          partner: '0xFa2c1bE677BE4BEc8851D1577B343F7060B51E3A',
+        },
+      },
+    ],
+  } as const
+}
+
+export function getV2RouterFilter(params: SwapActionParams) {
+  const { tokenIn, tokenOut, amountIn, amountOut, recipient } = params
+  const ethIn = tokenIn === zeroAddress
+  const tokenInUsed = tokenIn === zeroAddress ? WETH : tokenIn
+  const tokenOutUsed = tokenOut === zeroAddress ? WETH : tokenOut
+  return {
+    $abi: V2_ROUTER_ABI,
+    _path: buildV2PathQueryWithCase('checksum', tokenInUsed, tokenOutUsed),
+    _amountIn: ethIn ? undefined : amountIn,
+    _minOut: amountOut,
+    _receiver: recipient,
+  } as const
+}
+
+export function getHPSM2Filter(params: SwapActionParams) {
+  // amount is only for amountIn
+  const { tokenIn, tokenOut, amountIn } = params
+  const tokenInAddress = tokenIn ? getAddress(tokenIn) : undefined
+  const tokenOutAddress = tokenOut ? getAddress(tokenOut) : undefined
+
+  let inputs = {}
+
+  if (tokenIn && tokenOut) {
+    inputs = {
+      fxTokenAddress: { $or: [tokenInAddress, tokenOutAddress] },
+      peggedTokenAddress: { $or: [tokenInAddress, tokenOutAddress] },
+    }
+  } else if (tokenIn || tokenOut) {
+    const address = tokenInAddress || tokenOutAddress
+    inputs = {
+      $or: [{ fxTokenAddress: address }, { peggedTokenAddress: address }],
+    }
+  }
+
+  return {
+    $abi: HPSM2_ABI,
+    amount: amountIn,
+    ...inputs,
+  } as const
+}
+
+export function getHlpCurveV2Filter(params: SwapActionParams) {
+  const { tokenIn, tokenOut, amountIn, amountOut, recipient } = params
+  const minOut = tokenOut ? amountOut : undefined
+  if (tokenIn === zeroAddress) {
+    return {
+      $abi: HLP_CURVE_V2_ABI,
+      tokenOut,
+      minOut,
+      receiver: recipient,
+    }
+  }
+  return {
+    $abi: HLP_CURVE_V2_ABI,
+    tokenOut,
+    amountIn: tokenIn ? amountIn : undefined,
+    minOut,
+    receiver: recipient,
+    $or: [
+      {
+        peggedToken: tokenIn,
+      },
+      {
+        hlpToken: tokenIn,
+      },
+    ],
+  }
+}
+
+export function getHlpBalancerFilter(params: SwapActionParams) {
+  const { tokenIn, tokenOut, amountIn, amountOut, recipient } = params
+  const minOut = tokenOut ? amountOut : undefined
+  const ethUsedIn = tokenIn === zeroAddress
+  const tokenInput = ethUsedIn ? { tokenOut } : { tokenIn }
+  return {
+    $abi: HLP_BALANCER_ABI,
+    minOut,
+    amountIn: tokenIn && !ethUsedIn ? amountIn : undefined,
+    receiver: recipient,
+    ...tokenInput,
+  }
+}
+
+export function getCurveV2FactoryFilter(params: SwapActionParams) {
+  const { tokenIn, tokenOut, amountIn, amountOut } = params
+
+  // There isnt really a good way to tell which pool you are in without knowing the contract address of the pool
+  // i will always be fxUSD (0)
+  // j can either be FRAX, USDC.e, or USDT (may get false posiitves here)
```

Comment:
I think the false positives are fine - it's a pretty obscure edge-case. We'll find out if there's a way for the users to leverage this though, if there's a way they'll find it 😆

Reply:
Very obscure. The fxUSD frax pool sees roughly 1 tx a day, and the 3pool sees maybe 1 per week.